### PR TITLE
feat: added pseudo chat config; added wrapper for Bedrock client

### PR DESCRIPTION
**Pull request by adubovik**
Prerequisite for https://github.com/epam/ai-dial-adapter-bedrock/pull/35

### CODE REVIEW

File: aidial_adapter_bedrock/utils/concurrency.py [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
-from typing import Callable, TypeVar
+from typing import AsyncIterator, Callable, Iterator, Optional, Tuple, TypeVar
 
 T = TypeVar("T")
-A = TypeVar("A")
 
 
-async def make_async(func: Callable[[A], T], arg: A) -> T:
+async def make_async(func: Callable[[], T]) -> T:
     with ThreadPoolExecutor() as executor:
```
Comment (Oleksii-Klimov) [inline aidial_adapter_bedrock/utils/concurrency.py]: Maybe it's worth passing max_workers=1?

File: aidial_adapter_bedrock/utils/concurrency.py [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
+            return False, next(iter)
+        except StopIteration:
+            return True, None
+
+    while True:
+        is_end, item = await make_async(lambda: _next())
+        if is_end:
+            break
+        else:
+            assert item is not None
```
Comment (Oleksii-Klimov) [inline aidial_adapter_bedrock/utils/concurrency.py]: This validation doesn't seem to be necessary.

File: aidial_adapter_bedrock/utils/concurrency.py (hunk 1 above)
Comment (adubovik) [reply aidial_adapter_bedrock/utils/concurrency.py]: Fixed

File: aidial_adapter_bedrock/utils/concurrency.py (hunk 2 above)
Comment (adubovik) [reply aidial_adapter_bedrock/utils/concurrency.py]: Indeed. Added `cast`.

**Review by Oleksii-Klimov — APPROVED** [review]

**Pull request merged**